STN-343 : Implement the Traditional hero and carousel styles

heatherdesigns opened the pull request:
# [STN-343](https://sparkbox.atlassian.net/secure/RapidBoard.jspa?rapidView=80&modal=detail&selectedIssue=STN-343) READY FOR REVIEW

## Summary
Create the Hero and Carousel for the Traditional theme. They share the same functionality, behaviors, and use-cases as the colorful theme hero and carousel but with different styling.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. In the CLI, run `npm run test` and make sure all of our Sass tests pass.
2. Review the styles for Traditional hero and carousel components when they are placed within the top of the page and when they are placed with the main content of the page. Make sure they match the design comps: https://www.figma.com/file/bG42fKKioEcnJ0ORKkLL5E/⚙%EF%B8%8F-Traditional-Theme---Working?node-id=5686%3A2
3. The QA Page http://economics.suhumsci.loc/qa/qa-page has a couple implementations of the hero and carousel. I recommend creating an additional page to test various scenarios such as:
* An image, heading, copy, and link
* An image, heading, and copy
* An image and heading
* No image with a heading, copy, and/or link 
4. Review the Colorful hero and carousel and verify nothing has changed.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
feat: traditional hero styles

Commit:
feat: prep carousel for traditional work

- update all hb-colorful references to the hb-theme mixin so that when the airy theme is turned on, the carousel's will receive the colorful theme styles

Commit:
feat: add styles for traditional carousel

Commit:
feat: implement design feedback on hero

Commit:
feat: continue working on traditional carousel styles

Commit:
feat: wrap traditional carousel styles

Commit:
feat: design pairing style updates

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -5,7 +5,7 @@
 
   .hb-hero-overlay {
-    @include hb-colorful {
+    @include hb-themes(('colorful', 'airy')) {
```
good catch!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -136,11 +136,6 @@
           @include hb-heading-4;
           margin: hb-calculate-rems(24px) 0;
 
-          @include grid-media-min('md') {
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -2,20 +2,18 @@
 // If you change something here, you will likely need to change it and override it in the carousel.scss
 .field-hs-hero-image {
   // For the humsci_colorful theme hero, ensure there is a background color added to the image alone
```
edit this comment since it appears these styles now apply to multiple themes!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -40,33 +38,47 @@
-  @include hb-colorful {
-    position: relative;
-    min-height: 60vh;
+  position: relative;
```
can you move these styles up by the classname (hb-hero-overlay) - just seems a bit odd sitting in the middle here

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -40,33 +38,47 @@
-      width: 100%;
-      height: 100%;
-      z-index: $hb-z-index-hero-fallback;
+  .hb-three-column--one-sidebar & {
```
cool!

silverli replied:
what about for two-sidebars?

heatherdesigns replied:
I created a page to see what happens to the hero and carousel when there are two sidebars. The hero still looks great. :tada: But the carousel needed some love. The min-height of the image was fine but the content was swished. So I adjusted the margins and widths of content when there are two sidebars. :smile:

silverli replied:
looks great!!!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -76,123 +88,206 @@
         }
       }
+
+      .hb-three-column--one-sidebar & {
```
yeah maybe we need to consider two sidebar as well

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -76,123 +88,206 @@
+    left: 50%;
+    @include hb-page-width;
+
+    @include hb-traditional {
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -76,123 +88,206 @@
+      }
+    }
+
+    // must increase specificity of h2 to override line-height value set in hb-heading mixins
```
hmmm then why do the styles on line 134-147 work?

heatherdesigns replied:
I removed the `.field-hs-hero-title` class and added the correct `line-height` value for traditional themes to all the breakpoints. Colorful and traditional use the same property values set in the `hb-heading` mixins except for `line-height`.

silverli replied:
that makes sense to me!!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_mixins.buttons.scss:
```diff
@@ -210,22 +210,50 @@
+    }
+  }
+
+  @include hb-traditional {
```
🎉 looks good!!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -164,6 +159,85 @@
+
+        h2 {
+          width: auto;
+          margin: 0 hb-calculate-rems(32px);
```
you can use a hb-spacing-width variable here

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -164,6 +159,85 @@
+        }
+
+        .field-hs-hero-body {
+          margin: hb-calculate-rems(14px) hb-calculate-rems(32px) 0;
```
oo lots of hb-spacing-width you can use here :)

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -164,6 +159,85 @@
+        }
+
+        .field-hs-hero-link {
+          margin: 0 hb-calculate-rems(32px);
```
hb-spacing-width!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -164,6 +159,85 @@
+            width: calc(80% - #{hb-calculate-rems(140px)});
+          }
+
+          .hb-three-column--one-sidebar & {
```
may want to think about two sidebar here too

silverli reviewed (changes requested):
This is looking so good!! (and colorful still looks good!) I just have a couple comments I left in your code.

Here's some screenshots I got to record the different scenarios of hero and carousel:

<img width="895" alt="Screen Shot 2020-05-12 at 9 42 04 AM" src="https://user-images.githubusercontent.com/12848168/81703001-07e9d380-943a-11ea-8c71-37c0b5145d75.png">
<img width="905" alt="Screen Shot 2020-05-12 at 9 43 46 AM" src="https://user-images.githubusercontent.com/12848168/81703016-0c15f100-943a-11ea-931b-53280a09edfe.png">
<img width="896" alt="Screen Shot 2020-05-12 at 9 44 06 AM" src="https://user-images.githubusercontent.com/12848168/81703021-0e784b00-943a-11ea-9698-49d6c144b64f.png">

Commit:
feat: implement review feedback

- add styles for carousel when 2 sidebars are present
- update hero and carousel to use hb-spacing-width rather than hb-calculate-rems for greater future flexibility

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -1,21 +1,19 @@
-      opacity: 0.9;
-      z-index: $hb-z-index-hero1-overlay;
-    }
+  // Ensure there is a background color added to the image alone
```
👍

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.hero.scss:
```diff
@@ -34,39 +32,53 @@
 .hb-hero-overlay {
   margin: hb-spacing-width() 0 hb-spacing-width();
 
+  position: relative;
```
thanks!!

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/components/_pattern.carousel.scss:
```diff
@@ -175,36 +175,56 @@
         .field-hs-hero-link {
-          margin: 0 hb-calculate-rems(32px);
-
-          @include grid-media-min('md') {
```
did you mean to get rid of this media query?

silverli reviewed (approved):
thanks for considering the 2 sidebars!! it looks amazing. I completely and totally approve this PR :) 🎉 
<img width="1340" alt="Screen Shot 2020-05-12 at 12 23 41 PM" src="https://user-images.githubusercontent.com/12848168/81719902-aaf71900-944b-11ea-973d-114e20e98806.png">
<img width="1035" alt="Screen Shot 2020-05-12 at 12 24 02 PM" src="https://user-images.githubusercontent.com/12848168/81719915-afbbcd00-944b-11ea-9194-612d4aabb4fa.png">

Commit:
feat: add compiled CSS
